_includes/markdown/PHP.md

```diff
@@ -916,3 +916,98 @@ Instead of sessions, use cookies or client-side storage APIs if possible. In add
 If sessions must be used, create them conservatively. Don't create sessions for every visitor. Limit sessions to the smallest group that needs them: logged-in editors and admins, or visitors using a particular feature.
 
 Sessions should never be stored in the database. This introduces extra data into a storage system that's not meant for that volume. Database session libraries also rely on PHP code which can't match the performance of PHP's native session handlers. PHP extensions for Memcache and Redis allow sessions to be stored in these in-memory datastores and are a good solution for sessions when multiple webservers are present.
+
+### Avoid nested logic
+
+Nesting logic makes code hard to read. The following is an example of the [wp_unslash](https://codex.wordpress.org/Function_Reference/wp_unslash) function in WP Core.
+
+```php
+function wp_slash( $value ) {
+    if ( is_array( $value ) ) {
+        foreach ( $value as $k => $v ) {
+            if ( is_array( $v ) ) {
+                $value[ $k ] = wp_slash( $v );
+            } else {
+                $value[ $k ] = addslashes( $v );
+            }
+        }
+    } else {
+        $value = addslashes( $value );
+    }
+ 
+    return $value;
+}
+```
+
+This function could be rewritten to make it easier to understand and remove nested logic.
+
+```php
+function wp_slash( $value ) {
+  // Bail early if value is not an array.
+  if ( ! is_array( $value ) ) {
+    return addslashes( $value );
+  }
+  
+  // Loop through values and add slashes to each value.
+  foreach ( $value as $k => $v ) {
```

Comment:
I wasn't suggesting the _return value_ be cast, instead that the first conditional block is unnecessary if we coerce the `$value` parameter to always be an array 😉

Reply:
right - I mean if we skip the first block the return for $array = [
  0 => 'test',
]

would be

$array = [
  0 => [
  0 => 'test',
  ]
];

Reply:
You're right, I didn't realize it returns the same type its given - the first block is necessary there.

One thing I did catch is that `$values` needs to be initialized or handled in the return at the end to avoid an undefined variable notice if the function is called with an empty array.

Reply:
:derp: let me fix that real quick